### ISSUE TITLE
Enhance a11y for `PinInput`

### ISSUE THREAD

**Issue by hirotomoyamada** [yamada-ui/yamada-ui#1246]
path: stories/components/forms/pin-input.stories.tsx

### Description
1 accessibility violation was found in the `PinInput`.

#### [isReadonly](https://yamada-ui.github.io/yamada-ui/?path=/story/components-forms-pininput--is-readonly)

Expected the HTML found at $('section > .css-nqtsdw.ui-pin-input[aria-readonly="true"]') to have no violations:
    <div class="ui-pin-input css-nqtsdw" readonly="" aria-readonly="true" data-readonly="">

    Received:
    Elements must only use supported ARIA attributes (aria-allowed-attr)

    Fix all of the following:
      ARIA attribute is not allowed: aria-readonly="true"

    You can find more information on this issue here: 
    https://dequeuniversity.com/rules/axe/4.9/aria-allowed-attr?application=axeAPI

Expected the HTML found at $('.ui-form__control > .css-nqtsdw.ui-pin-input[aria-readonly="true"]') to have no violations:
    <div class="ui-pin-input css-nqtsdw" readonly="" aria-readonly="true" data-readonly="">

    Received:
    Elements must only use supported ARIA attributes (aria-allowed-attr)

    Fix all of the following:
      ARIA attribute is not allowed: aria-readonly="true"

    You can find more information on this issue here: 
    https://dequeuniversity.com/rules/axe/4.9/aria-allowed-attr?application=axeAPI

### Problem Statement/Justification
A11y is essential to ensure quality when operating a package.

### Additional Information
While it's ideal to cover all jobs, if there are too many jobs to work on, please create a PR within your capacity.
Any deficiencies not covered by the PR will be automatically created as issues.

**Comment by GensIto** [comment]
@hirotomoyamada 
can I work on this issue??

**Comment by hirotomoyamada** [comment]
@GensIto 

Hi, thanks for the cooment.

Yes, please.

I'll assign this issue to you. Good luck!
